### DISCUSSION TITLE
anyway to use JUST ttgo 2 buttons for menus?

### DISCUSSION THREAD

**Discussion by emaayan** [davetcc/tcMenu#363] [Q&A] [answered]
hi..
the ttgo has 2 buttons in gpio 0 and 35, is there a way i can specify one of them can be used as OK (select) button ?
the generator only allows for right or left buttons to be used as ok.

**Comment by davetcc** [comment]
If you mean can you control a menu with just two buttons, not out of the box, this has been discussed before, and I think people have written their own custom encoders based around the information here https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menumanager-and-iteration/#controlling-the-menu-items-manually and looking at the existing RotaryEncoder implementations in switches.

**Reply by davetcc** [reply]
I think there are more questions here about how you'd control a menu with only two buttons, I'm not even sure how user-friendly it would be. How would you move through the items in the menu if you only had two buttons?

Like I say, I think others have done this before by making compromises somewhere, one such compromise was to have a single button to cycle through items and another button acting as select. It is not difficult to come up with an implementation of RotaryEncoder that did that, and it may even work for very small menus. 

If a proper and viable way of it working could be defined that was user-friendly enough to actually work, I'd incorporate it into 4.1.

**Comment by emaayan** [comment]
Well, the  pine soldering irons have 2 buttons on them

I think a good way is to simply have a long press on up button to select
menu entry and long press on down button to exit menu entry.

**Reply by davetcc** [reply]
Given it seems to have been upvoted a few times, I think that shouldn't be a lot of work, I'll add it in 4.1.

**Reply by davetcc** [reply]
https://github.com/davetcc/tcMenu/issues/370

**Reply by davetcc** [reply]
See the above issue, will be released in 4.1

**Reply by davetcc** [reply, marked as answer]
This should now be available in 4.1
